xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1424,6 +1424,84 @@ HDR_STATUS CWIN32Util::GetWindowsHDRStatus()
   return status;
 }
 
+/*!
+ * \brief Retrieve from the system the max luminance of SDR content in HDR.
+ * \param GdiDeviceName The screen to retrieve the information for
+ * \param SDRWhiteLevel Max luminance in nits
+ * \return true if a value could be read from the system, false otherwise
+*/
+bool CWIN32Util::GetSystemSdrWhiteLevel(const std::wstring& gdiDeviceName, unsigned int* sdrWhiteLevel)
+{
+#ifdef TARGET_WINDOWS_STORE
+  // Not supported
+  return false;
+#else
+  // Support of DISPLAYCONFIG_DEVICE_INFO_GET_SDR_WHITE_LEVEL started in Windows 10 1709
+  // Do earlier OS versions fail gracefully in DisplayConfigGetDeviceInfo()
+  // or does the OS version have to be confirmed before calling?
+
+  uint32_t pathCount{0};
+  uint32_t modeCount{0};
+  std::vector<DISPLAYCONFIG_PATH_INFO> paths;
+  std::vector<DISPLAYCONFIG_MODE_INFO> modes;
+
+  uint32_t flags = QDC_ONLY_ACTIVE_PATHS;
+  LONG result = ERROR_SUCCESS;
+
+  do
+  {
+    if (GetDisplayConfigBufferSizes(flags, &pathCount, &modeCount) != ERROR_SUCCESS)
+      return false;
+
+    paths.resize(pathCount);
+    modes.resize(modeCount);
+
+    result = QueryDisplayConfig(flags, &pathCount, paths.data(), &modeCount, modes.data(), nullptr);
+  } while (result == ERROR_INSUFFICIENT_BUFFER);
+
+  if (result == ERROR_SUCCESS)
+  {
+    paths.resize(pathCount);
+    modes.resize(modeCount);
+
+    for (const auto& path : paths)
+    {
+      DISPLAYCONFIG_SOURCE_DEVICE_NAME source{};
+      source.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
+      source.header.size = sizeof(source);
+      source.header.adapterId = path.sourceInfo.adapterId;
+      source.header.id = path.sourceInfo.id;
+
+      if (DisplayConfigGetDeviceInfo(&source.header) == ERROR_SUCCESS)
+      {
+        if (gdiDeviceName == source.viewGdiDeviceName)
+        {
+          DISPLAYCONFIG_SDR_WHITE_LEVEL config{};
+          config.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SDR_WHITE_LEVEL;
+          config.header.size = sizeof(config);
+          config.header.adapterId = path.targetInfo.adapterId;
+          config.header.id = path.targetInfo.id;
+          if (DisplayConfigGetDeviceInfo(&config.header) == ERROR_SUCCESS)
+          {
+            if (sdrWhiteLevel)
+            {
+              ULONG nits = config.SDRWhiteLevel * 80 / 1000;
+              if (nits <= std::numeric_limits<unsigned int>::max())
```

Comment:
This seems too complicated, maybe:

```c++
if (nits > 10000)
  *sdrWhiteLevel = 10000U;
else
 *sdrWhiteLevel = static_cast<unsigned int>(nits);
```

Reply:
yes, might as well clamp the nits at the same time.